File: 1342-number-of-steps-to-reduce-a-number-to-zero/1342-number-of-steps-to-reduce-a-number-to-zero.js
/**
 * @param {number} num
 * @return {number}
 */
var numberOfSteps = function (num) {
  let res = 0;
  while (num !== 0) {
    if (num % 2 !== 0) {
      num -= 1;
    } else {
      num /= 2;
    }
    res += 1;
  }
  return res;
};

File: 1342-number-of-steps-to-reduce-a-number-to-zero/1342-number-of-steps-to-reduce-a-number-to-zero.cpp
class Solution {
public:
    int numberOfSteps(int num) {
        int res=0;
        while (num!=0)
        {
            if (num%2==0) {
                num=num/2;
            } else {
                num-=1;
            }
            res+=1;
        }
        return res;
    }
};